lib/poseidon/connection.rb

```diff
@@ -113,7 +113,11 @@ def ensure_connected
 
     # Explained on http://spin.atomicobject.com/2013/09/30/socket-connection-timeout-ruby/
     def connect_with_timeout(host, port, timeout = 5)
```

Comment:
See comments in the noted article...looks like we weren't the only ones to run into issues with resolving the host in this implementation. ;-)